orca-whirlpool/src/modules/8_map_swaps.rs

```diff
@@ -0,0 +1,182 @@
+use crate::key_store::StoreKey;
+use crate::pb::messari::orca_whirlpool::v1::{event::Type, Event, Events, Pool, Swap, Swaps};
+use substreams::log;
+use substreams::{
+    skip_empty_output,
+    store::{StoreGet, StoreGetProto},
+};
+
+#[substreams::handlers::map]
+pub fn map_swaps(
+    raw_events: Events,
+    pool_store: StoreGetProto<Pool>,
+) -> Result<Swaps, substreams::errors::Error> {
+    skip_empty_output();
+
+    let zero = "0".to_string();
+    let mut swaps: Vec<Swap> = Vec::new();
+
+    for event in raw_events.data {
+        if let Some(event_type) = event.r#type.clone() {
+            match event_type {
+                Type::TwoHopSwap(two_hop_swap_event) => {
+                    if let (Some(instruction), Some(accounts)) =
+                        (two_hop_swap_event.instruction, two_hop_swap_event.accounts)
+                    {
+                        let swap_a_to_b_one = handle_swap(
+                            instruction.a_to_b_one,
+                            instruction
+                                .amount_a_one
+                                .clone()
+                                .unwrap_or_else(|| zero.clone()),
```

Comment:
I'd like to see `zero` come from a constant.rs file for normal types and BigInts. You don't need to update here to merge, though.